block-producer/src/server/api.rs

```diff
@@ -32,15 +34,31 @@ impl<T> api_server::Api for BlockProducerApi<T>
 where
     T: TransactionQueue,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip(self, request), err)]
     async fn submit_proven_transaction(
```

Comment:
I think it may be OK to override the `target` field, but wouldn't setting component at the top-level span be simpler?

Also, I'm assuming we are using string literals here because we can't use constants? If so, shouldn't we use string literals everywhere (instead of sometimes using string literals and sometimes using `target!()` macro)?